lib/MongoDB/BulkWrite.pm

```diff
@@ -259,6 +263,8 @@ sub execute {
         bypassDocumentValidation => $self->bypassDocumentValidation,
         bson_codec               => $self->collection->bson_codec,
         write_concern            => $write_concern,
+        client                   => $self->_client,
+        ( defined $self->session ? ( session => $self->session ) : () ),
```

Comment:
This should come from an options hash given to `execute` rather than an attribute.  It should resemble how things work in Collection.pm.